src/plugins/rust/templates/tokenizer.template.rs

```diff
@@ -143,29 +151,42 @@ impl Tokenizer {
             .get(self.get_current_state())
             .unwrap();
 
-        for i in 0..lex_rules_for_state.len() {
-            let lex_rule = LEX_RULES[i];
+        let mut max_match_len = 0;
+        let mut max_match_token: Option<&'static str> = None;
+
+        for i in lex_rules_for_state {
+            let i = *i as usize;
 
-            if let Some(matched) = self._match(str_slice, &Regex::new(lex_rule).unwrap()) {
+            // the previous author use this to generate a new regex expression
+            // every time when `get_next_token` is called
+            // now this variable is of no use, I leave it here for debug use
+            let _lex_rule = LEX_RULES[i];
```

Comment:
Just remove this whole comment, and the `_lex_rule` (I used it mostly for readability).